server/src/main/java/io/crate/expression/tablefunctions/GenerateSubscripts.java

```diff
@@ -0,0 +1,188 @@
+            ).withTypeVariableConstraints(typeVariable("E")),
+            GenerateSubscripts::new
+        );
+        module.register(
+            Signature.table(
+                NAME,
+                parseTypeSignature("array(E)"),
+                DataTypes.INTEGER.getTypeSignature(),
+                DataTypes.BOOLEAN.getTypeSignature(),
+                DataTypes.INTEGER.getTypeSignature()
+            ).withTypeVariableConstraints(typeVariable("E")),
+            GenerateSubscripts::new
+        );
+    }
+
+    private final Signature signature;
+    private final Signature boundSignature;
+
+    private GenerateSubscripts(Signature signature, Signature boundSignature) {
+        this.signature = signature;
+        this.boundSignature = boundSignature;
+    }
+
+    private static List<?> extractNestedArray(List<?> array, int targetLevel) {
+        List<?> targetArray = array;
+        for (int level = 2; level <= targetLevel; level++) {
+            if (targetArray == null || targetArray.isEmpty()) {
+                return null;
+            }
+            targetArray = checkAllElementsAreArraysOfConsistentDimension(targetArray, level);
+        }
+        return targetArray;
+    }
+
+    private static List<?> checkAllElementsAreArraysOfConsistentDimension(List<?> array, int level) {
+        int dim = -1;
+        List<?> firstNonNullElement = null;
+        for (int i = 0; i < array.size(); i++) {
+            Object oi = array.get(i);
+            if (oi != null && !(oi instanceof List)) {
+                throw new IllegalArgumentException(String.format(
+                    Locale.ENGLISH,
+                    "expected a nested array, or null value at level %d, position %d",
+                    level, i + 1));
+            }
+            if (oi == null) {
+                // null is a valid value within an array
+                continue;
+            }
+            List<?> element = (List<?>) oi;
+            if (dim == -1) {
+                dim = element.size();
+            } else {
+                if (dim != element.size()) {
+                    throw new IllegalArgumentException(String.format(
+                        Locale.ENGLISH,
+                        "nested arrays must have the same dimension within a level, offending level %d, position %d",
+                        level, i + 1));
+                }
+            }
+            if (firstNonNullElement == null) {
+                firstNonNullElement = element;
+            }
+        }
+        return firstNonNullElement;
+    }
+
+    @Override
+    public Iterable<Row> evaluate(TransactionContext txnCtx, Input<T>... args) {
+        assert args.length == 2 || args.length == 3 :
+            "Signature must ensure that there are either two or three arguments";
+
+        List<?> array = (List<?>) args[0].value();
+        if (array == null || array.isEmpty()) {
+            return () -> Collections.emptyIterator();
```

Comment:
```suggestion
            return List.of();
```

No need to create a special supplier/`Iterable`.